diff --git a/tools/Makefile b/tools/Makefile
index 78c366b..79c49be 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -71,6 +71,7 @@ tools-y += xz
 tools-y += zip
 tools-y += zlib
 tools-y += fdt-patch-dm-verify
+tools-y += ar-tool
 tools-$(if $(CONFIG_JSON_CHECKSEC_MITIGATIONS_REPORT),y) += checksec.sh
 tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS),y) += liblzo
 tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(BUILD_B43_TOOLS),y) += b43-tools
