diff --git a/include/kernel-6.6 b/include/kernel-6.6
new file mode 100644
index 0000000..2e70a47
--- /dev/null
+++ b/include/kernel-6.6
@@ -0,0 +1,2 @@
+LINUX_VERSION-6.6 = .79
+LINUX_KERNEL_HASH-6.6.79 = 07a6f904470da1a099aa1683e3025a999dd82f2438f78b006b80c6ae2e9dfe8d
diff --git a/include/target.mk b/include/target.mk
index 6514c53..92caaae 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -143,11 +143,11 @@ ifneq ($(TARGET_BUILD)$(if $(DUMP),,1),)
   include $(INCLUDE_DIR)/kernel-version.mk
 endif
 
-GENERIC_PLATFORM_DIR := $(TOPDIR)/target/linux/generic
-GENERIC_BACKPORT_DIR := $(GENERIC_PLATFORM_DIR)/backport$(if $(wildcard $(GENERIC_PLATFORM_DIR)/backport-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
-GENERIC_PATCH_DIR := $(GENERIC_PLATFORM_DIR)/pending$(if $(wildcard $(GENERIC_PLATFORM_DIR)/pending-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
-GENERIC_HACK_DIR := $(GENERIC_PLATFORM_DIR)/hack$(if $(wildcard $(GENERIC_PLATFORM_DIR)/hack-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
-GENERIC_FILES_DIR := $(foreach dir,$(wildcard $(GENERIC_PLATFORM_DIR)/files $(GENERIC_PLATFORM_DIR)/files-$(KERNEL_PATCHVER)),"$(dir)")
+GENERIC_PLATFORM_DIR ?= $(TOPDIR)/target/linux/generic
+GENERIC_BACKPORT_DIR ?= $(GENERIC_PLATFORM_DIR)/backport$(if $(wildcard $(GENERIC_PLATFORM_DIR)/backport-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_PATCH_DIR ?= $(GENERIC_PLATFORM_DIR)/pending$(if $(wildcard $(GENERIC_PLATFORM_DIR)/pending-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_HACK_DIR ?= $(GENERIC_PLATFORM_DIR)/hack$(if $(wildcard $(GENERIC_PLATFORM_DIR)/hack-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_FILES_DIR ?= $(foreach dir,$(wildcard $(GENERIC_PLATFORM_DIR)/files $(GENERIC_PLATFORM_DIR)/files-$(KERNEL_PATCHVER)),"$(dir)")
 
 __config_name_list = $(1)/config-$(KERNEL_PATCHVER) $(1)/config-default
 __config_list = $(firstword $(wildcard $(call __config_name_list,$(1))))
diff --git a/package/base-files/Makefile b/package/base-files/Makefile
index ce659d8..828edf0 100644
--- a/package/base-files/Makefile
+++ b/package/base-files/Makefile
@@ -17,6 +17,7 @@ PKG_RELEASE:=$(COMMITCOUNT)
 
 PKG_FILE_DEPENDS:=$(PLATFORM_DIR)/ $(GENERIC_PLATFORM_DIR)/base-files/
 PKG_BUILD_DEPENDS:=usign/host ucert/host
+PKG_CONFIG_DEPENDS:=CONFIG_PACKAGE_wifi-scripts
 PKG_LICENSE:=GPL-2.0
 
 # Extend depends from version.mk
@@ -240,6 +241,8 @@ endif
 
 	$(if $(CONFIG_TARGET_PREINIT_DISABLE_FAILSAFE), \
 		rm -f $(1)/etc/banner.failsafe,)
+	$(if $(CONFIG_PACKAGE_wifi-scripts), \
+		rm -f $(1)/sbin/wifi,)
 endef
 
 ifneq ($(DUMP),1)
diff --git a/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c b/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
index 522085b..944ef7c 100644
--- a/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
+++ b/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c
@@ -538,9 +538,14 @@ static int gpio_keys_button_probe(struct platform_device *pdev,
 			struct device_node *child =
 				of_get_next_child(dev->of_node, prev);
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,6,1))
+			bdata->gpiod = devm_fwnode_gpiod_get(dev,
+				of_fwnode_handle(child), NULL, GPIOD_IN,
+				desc);
+#else
 			bdata->gpiod = devm_gpiod_get_from_of_node(dev,
 				child, "gpios", 0, GPIOD_IN, desc);
-
+#endif
 			prev = child;
 		}
 
diff --git a/package/network/config/netifd/Makefile b/package/network/config/netifd/Makefile
index acfabf0..a2d6d35 100644
--- a/package/network/config/netifd/Makefile
+++ b/package/network/config/netifd/Makefile
@@ -9,6 +9,7 @@ PKG_SOURCE_DATE:=2024-01-04
 PKG_SOURCE_VERSION:=c18cc79d50002ab8529c21184aceb016c61ac61c
 PKG_MIRROR_HASH:=0a1080ade51dc4a55249c8899d4d384f665e0d21075adab24ea23a2808165e05
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
+PKG_CONFIG_DEPENDS:=CONFIG_PACKAGE_wifi-scripts
 
 PKG_LICENSE:=GPL-2.0
 PKG_LICENSE_FILES:=
@@ -44,6 +45,8 @@ define Package/netifd/install
 	$(CP) ./files/* $(1)/
 	$(INSTALL_DIR) $(1)/etc/udhcpc.user.d/
 	$(CP) $(PKG_BUILD_DIR)/scripts/* $(1)/lib/netifd/
+	$(if $(CONFIG_PACKAGE_wifi-scripts), \
+		rm $(1)/lib/netifd/netifd-wireless.sh,)
 endef
 
 $(eval $(call BuildPackage,netifd))
diff --git a/profiles/mtk_filogic.yml b/profiles/mtk_filogic.yml
new file mode 100644
index 0000000..2c296eb
--- /dev/null
+++ b/profiles/mtk_filogic.yml
@@ -0,0 +1,51 @@
+---
+profiles:
+  - mediatek_mt7988a-arcadyan-mozart
+target: mediatek
+subtarget: filogic
+external_target: True
+description: Build image for MediaTek / Arcadyan Mozart board
+feeds:
+  - name: feed_mediatek
+    uri: https://git01.mediatek.com/filogic/prolos/prplos-feed-mediatek
+    tracking_branch: master
+    revision: # <need to replace with commit hash>
+
+packages:
+  - mediatek
+  - kmod-mt7996-firmware
+  - kmod-mt7996-233-firmware
+  - kmod-mt7996e
+  - kmod-br-netfilter
+  - kmod-ipt-offload
+  - ucode-mod-uci
+  - netifd
+  - wifi-scripts
+  - wpa-cli
+  - hostapd-utils
+  - libudebug
+  - iptables
+  - iptables-mod-conntrack-extra
+  - iptables-mod-extra
+  - iptables-mod-ipopt
+  - mtkhnat_util
+  - mt7988-2p5g-phy-firmware
+  - mt7988-wo-firmware
+  - aqr10g-phy-firmware
+  - smp_util
+  - regs
+  - mt76-test
+  - tcpdump
+  - mpstat
+  - tftp
+  - wifi_shutdown_util
+  - netfilter-flowtable
+
+diffconfig: |
+  CONFIG_TARGET_MULTI_PROFILE=y
+  CONFIG_TARGET_DEVICE_mediatek_filogic_DEVICE_mediatek_mt7988a-rfb=y
+  CONFIG_TARGET_DEVICE_mediatek_filogic_DEVICE_mediatek_mt7988d-rfb=y
+  CONFIG_TARGET_DEVICE_mediatek_filogic_DEVICE_arcadyan_mozart=y
+  CONFIG_PACKAGE_CFG80211_TESTMODE=y
+  CONFIG_WPA_MSG_MIN_PRIORITY=0
+  # CONFIG_PACKAGE_wpad-basic-mbedtls is not set
diff --git a/scripts/gen_config.py b/scripts/gen_config.py
index 1554189..1f6fcaa 100755
--- a/scripts/gen_config.py
+++ b/scripts/gen_config.py
@@ -203,7 +203,7 @@ for ap in profile.get("additional_packages"):
         die(f"Error installing additional packages {packages_install} from {feed} feed")
 
 if profile.get("external_target", False):
-    if run_cmd(["./scripts/feeds", "install", profile["target"]]).returncode:
+    if run_cmd(["./scripts/feeds", "install", "-f", profile["target"]]).returncode:
         die(f"Error installing external target {profile['target']}")
 
 config_output = f"""CONFIG_TARGET_{profile["target"]}=y
-- 
2.45.2

